cmd/brew-upgrade-mysql

```diff
@@ -1,34 +1,55 @@
 #!/bin/bash
+#
+# This is a script to insure that the correct version of MySQL is
+# installed on dev environments. This is most useful during major
+# version upgrades. It is also useful if we need to update the developer
+# my.cnf file. It also ensures that `mysql_upgrade` has been run on the
+# MySQL servers so no unwanted behaviors happen.
+#
+# This can be run with "latest" as an option to upgrade to the latest
+# minor version on MySQL.
+#
 
 set -e
 
-upgrade_mysql() {
-  echo "Running mysql_upgrade..."
-
-  local exitstatus=0
-  # Hide failures from "already upgraded", otherwise let it bubble up
-  if ! output=$($(brew --prefix mysql@5.7)/bin/mysql_upgrade -u root); then
-    exitstatus="$?"
-    if echo "$output" | grep --quiet "already upgraded"; then
-      exitstatus=0
-    else
-      # If there's some other reason for this, we want to make sure it's heard
-      echo "$output"
+install_mysql() {
+  current_mysql=$(brew list | grep -E "^mysql(@\\d\\.\\d)?$")
+  if ! [[ "${current_mysql}" == "mysql" || "${current_mysql}" == "mysql@${mysql_version}" ]]; then
+    if [ -n "${current_mysql}" ]; then
+      echo -n "Uninstalling old version of MySQL... "
+      brew services stop $current_mysql &> /dev/null || true
+      brew uninstall $current_mysql &> /dev/null
+      echo "done"
     fi
+    echo -n "Installing new version of MySQL... "
+    brew install mysql@${mysql_version} &> /dev/null
+    echo "done"
+  elif [ "${dot_upgrade}" = "latest" ]; then
+    v=$(brew info mysql --json=v1 | jq -r ".[] | .versions.stable")
+    echo -n "Upgrading version of MySQL to ${v}... "
+    brew services stop mysql@${mysql_version} &> /dev/null || true
+    brew upgrade mysql@${mysql_version} &> /dev/null || true
```

Comment:
I would consider not hiding the output of all these commands because if something goes wrong (e.g. `brew link` fails) then users will not be made aware of that.